src/engine/renderer/glsl_source/lightMapping_fp.glsl

```diff
@@ -108,7 +108,13 @@ void main()
 	#endif // !r_normalMapping
 
 	// Compute the material term.
-	vec4 material = texture2D(u_MaterialMap, texCoords);
+	#if defined(USE_AUTOMATIC_SPECULARMAP)
+		float mean = (diffuse.r + diffuse.g + diffuse.b) * 0.333333;
+		mean = max(max(max(mean, diffuse.r), diffuse.g), diffuse.b);
```

Comment:
These two lines are just the same as taking the max component, i.e. `float mean = max(max(diffuse.r, diffuse.g), diffuse.b)`

Reply:
Oh, right! 😅️ Now fixed.